### ISSUE TITLE
Custom Code compil doesn't work anymore

### ISSUE THREAD

**Issue by JeanLouisPiquet78125** [FlutterFlow/flutterflow-issues#3091]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

My custom code include integration of PubDev PDF and Printing.. All was working fine till few minutes ago when compilation was not working anymore.
I tried to change the version of PDF and Printing to a newer version but still doesn't work.

### Expected Behavior

No expected behavior unless a normal compilation.

### Steps to Reproduce

Compilation of the custoom code generatePDF

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT40i+fl6ot2mNtF7c7qacEw+So5KWI/a+c3jMJ7aCgoIYyuOpl/evWkYhFuONe6YAhYM0Smmnsew86JjYbxMcI6Cz+bfKZc+LpqUxXMZFuue5OBBs+0RHYkHdNYI0yE1qW7ryNQBMRaR0Fg3EyIDe7IQj/DNufHIndISq/LZO4=

### Visual documentation

![Capture d’écran 2024-06-10 à 15 22 42](https://github.com/FlutterFlow/flutterflow-issues/assets/147150885/edfeca5b-40d7-40f2-92b2-153794230fad)

### Environment

```markdown
- MacOS Monterry v 12.7.5
- Chrome browserFlutterFlow version:
```

**Comment by msusviela** [comment]
Hi @JeanLouisPiquet78125<br><br>I wasn't able to reproduce your issue, even for your project.

Could you specify more info, like for example the platform you're trying to compile?

**Comment by JeanLouisPiquet78125** [comment]
Hi there,
Thanks for your answer.
As I need to move on my project, I went checking here and there to try to
understand and made some modifications since I raised the ticket.
If you have access to my project, there are 2 custom actions. One is called
generatePdf and the other one is uploadFileToBucket.
the ticket I raised (#3091 - Custom code compil doesn't work anymore) is
related to custom action generatePdf.
Basically the principle is: I create a PDF and then I display it using
PubDev printing: ^5.12.0.

Everything was working well and compiling nicely till a time when
compilation was crashing but I was just bringing some additions to the PDF.
Looks like the problem is coming from a dependency :dart:ffi which is not
available anymore ??? I don't really understand that point. Does this mean
that I must forget using Pbdev printing or is there any other sustainable
alternative ?

So the only solution I found ifor being able to keep on working on the
project s to disconnect the very last part of the code in generatePdf
custom action, i.e
// PRINT IT == using printing: ^5.12.0
// await Printing.layoutPdf(
// onLayout: (PdfPageFormat format) async => pdf.save());

However,  this is not solving my problem which is displaying the PDF on the
screen and be able to view it using iOS native viewer or printing it, etc...

In the meantime, I created another custom code which I need :
uploadFileToBucket in which the idea is: i) create the PDF, then ii) save
it
to Supabase storage bucket and iii) update an AppSate for retrieving the
file Supabase's URL and sending the URL via email (using as a first step
the sendEmail basic fultterflow action in this PoC and then, if PoC is
accepted to automatically send the PDF).

Here the code is compiling without any error. But nothing happens (means
when I trigger the action, the file is not saved in the bucket...).

So I feel so dumb... Is there any solution for displaying the PDF ? What is
wrong with the code for uploading the PDF ?

Looking forward to hearing from you,
Kind regards,







 is for
